app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java

```diff
@@ -405,15 +395,23 @@ public static void unBondIfBlukonAtInit() {
                 gotLowBat = true;
             }
 
-            if (JoH.pratelimit(BLUKON_GETSENSORAGE_TIMER, GET_SENSOR_AGE_DELAY)) {
-                currentCommand = GET_SENSOR_TIME_COMMAND;
-                Log.i(TAG, "getSensorAge");
+            /* LibreAlarmReceiver.CalculateFromDataTransferObject, called when processing historical data,
+             * expects the sensor age not to be updated yet, so only update the sensor age when not retrieving history.
+             */
+            if (Pref.getBooleanDefaultFalse("external_blukon_algorithm") || getHistoricReadings) {
+                // Send the command to getHistoricData (read all blcoks from 0 to 0x2b)
```

Comment:
Since you're moving things around, recommend changing "blcoks" to "blocks".

Reply:
done